Loading comps from lolchess.gg

Sizzzles opened the pull request:
I originally found this version from a closed pull request and tried my best to revive it. I know you may not want this version to replace the original bot and I agree. I think this version would be best on a secondary branch that way the original can always be used for those that don't want automated comps but sadly I don't have the option to do that

zxcvqwerasdf committed:
release

zxcvqwerasdf committed:
readme update

Sizzzles committed:
level up in early rounds

Sizzzles committed:
Fix misspelled item and augment position update

Sizzzles committed:
Fixed warnings

Sizzzles committed:
Updated for set 8

Sizzzles committed:
Comp importing works again

Sizzzles committed:
Checks items more frequently and added more augments

Sizzzles committed:
Added more level ups

STAGE  2-1     LEVEL  4
STAGE: 2-5     LEVEL  5
STAGE: 3-2     LEVEL  6
STAGE: 4-1     LEVEL  7
STAGE: 5-1     LEVEL  8

Sizzzles committed:
Exits game after victory instead of waiting

Sizzzles committed:
Augment reroll and spacing cleanup

Sizzzles committed:
Clears anvils after PvE rounds

Sizzzles committed:
Delay for champion moves and Pillow ver update

Sizzzles committed:
Update screen_coords.py

Sizzzles committed:
updated screen_coords.py

Sizzzles committed:
Updated item pickup for units put  on far right

Sizzzles committed:
Victory continue button actually works now

Sizzzles committed:
Added victory exit fuction, Should work now

Sizzzles committed:
Faster accept on queues, should pickup all orbs

Sizzzles committed:
Actually selects augments from list

Sizzzles committed:
Update arena.py

Sizzzles committed:
Update screen_coords.py

Sizzzles committed:
Changed league path back to default

Sizzzles committed:
Fixed Augments not being picked at 2-1

Sizzzles committed:
Updated coords for comps such as underground

Sizzzles committed:
Item placement now happens after anvil rounds

Sizzzles committed:
Update README.md

Sizzzles committed:
Code Cleanup

Sizzzles committed:
Fixed Auto Comps

Sizzzles committed:
auto comps rewrite || New requirements

Sizzzles committed:
Added planned feature

Sizzzles committed:
Dynamic augments testing

Sizzzles committed:
Dynamic augments testing

Sizzzles committed:
Update README.md

Sizzzles committed:
Dynamic and user defined augments

Sizzzles committed:
Dynamic and user defined augments

Sizzzles committed:
Merge branch 'testing' of https://github.com/Sizzzles/TFT-OCR-BOT into testing

Sizzzles committed:
Dynamic and user defined augments

Auto comps now have augments for each comp. If augment isn't found defaults to user defined augment then if still not found defaults to first augment position

Sizzzles committed:
Delete cached_data directory

Sizzzles committed:
removed unneeded default position after carousel

Sizzzles committed:
Added missing buy champ 3 when bench full

Sizzzles committed:
Merge branch 'main' of https://github.com/Sizzzles/TFT-OCR-BOT

Sizzzles committed:
Merge branch 'main' into main

Sizzzles committed:
Update arena.py

Arborym commented:
Just tried this out and received this error:
```txt
Traceback (most recent call last):
  File "c:\Users\..\..\..\..\TFT-OCR-BOT\main.py", line 33, in <module>
    auto_comps.LoadChampionsAndComps(comps_manager)
  File "c:\Users\..\..\..\..\TFT-OCR-BOT\auto_comps.py", line 160, in LoadChampionsAndComps
    lol_chess_comps = __LoadLolChessComps(response_meta_comps, set_current, comp_manager)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\dpari\Desktop\GitHub Projects\Public\TFT-OCR-BOT\auto_comps.py", line 91, in __LoadLolChessComps
    real_augments = [arg.get("name") for arg in query_data.get("augments", []) if arg.get("key") in augments]
                                                ^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
```

Sizzzles commented:
Noted. Will fix when I can

Sizzzles committed:
AttributeError fix

Sizzzles commented:
Should work now

Arborym commented on .gitignore:
```diff
@@ -334,3 +334,6 @@ ASALocalRun/
 .vscode
+test2222.py
+dump_comps.py
+cached_data/cached8.json
```
```suggestion
cached_data/cached8.json
cached_data/deck.json
cached_data/inputed

```

Arborym commented on arena.py:
```diff
@@ -326,20 +328,35 @@ def buy_xp_round(self) -> None:
             mk_functions.buy_xp()
 
+
+    def load_aguments(self):
+        augments = self.comps_manager.CURRENT_COMP()[2]
+        return augments
```
Returning an immediate inline variable
```suggestion
    def load_aguments(self):
        return self.comps_manager.CURRENT_COMP()[2]
```

Arborym commented on arena.py:
```diff
@@ -326,20 +328,35 @@ def buy_xp_round(self) -> None:
+                        print(f"  Choosing comp defined augment: {augment}")
+                        mk_functions.left_click(screen_coords.AUGMENT_LOC[augments.index(augment)].get_coords())
+                        return
+            except:
```
```suggestion
            except Exception:
```

Arborym commented on arena_functions.py:
```diff
@@ -43,34 +43,29 @@ def get_gold() -> int:
-        ),
-        "",
-    )
+    for champion in comps.champions:
+        if SequenceMatcher(a=champion, b=champ).ratio() >= 0.7:
+            return champion
+    return ""
```
Respective of the logic I implemented a while ago on the main branch, I think it's best to keep the original version of this
```suggestion
    return next(
        (
            champion
            for champion in comps.champions
            if SequenceMatcher(a=champion, b=champ).ratio() >= 0.7
        ),
        "",
    )
```

Arborym commented on arena_functions.py:
```diff
@@ -104,15 +99,10 @@ def bench_occupied_check() -> list:
-        ),
-        None,
-    )
+    for valid_item_name in game_assets.ITEMS:
+        if valid_item_name in item or SequenceMatcher(a=valid_item_name, b=item).ratio() >= 0.7:
+            return valid_item_name
+    return None
```
Same as above
```suggestion
    return next(
        (
            valid_item_name
            for valid_item_name in game_assets.ITEMS
            if valid_item_name in item
            or SequenceMatcher(a=valid_item_name, b=item).ratio() >= 0.7
        ),
        None,
    )
```

Arborym commented on auto_comps.py:
```diff
@@ -0,0 +1,224 @@
+    json_in_text = Parse(response.text, 'window.guideChampion = {\n            champions: ', r'}}],')[1] + r'}}]'
+    json_parsed = json.loads(json_in_text)
+
+    output_dictionary = {}
+    for each_character in json_parsed:
+        character_code = each_character['code']
+        character_name_ingame = each_character['ingame_code']
+        character_price = int(each_character['cost'])
+        if character_name_ingame not in output_dictionary:
+            output_dictionary[character_name_ingame] = (character_price, character_code)
+    return [current_tft_set, output_dictionary]
```
Some swapping/clean up here
```suggestion
    output_dictionary = {}
    for each_character in json_parsed:
        character_code = each_character['code']
        character_name_ingame = each_character['ingame_code']
        character_price = int(each_character['cost'])
        if character_name_ingame not in output_dictionary:
            character_code = each_character['code']
            character_price = int(each_character['cost'])
            output_dictionary[character_name_ingame] = (character_price, character_code)
    return [current_tft_set, output_dictionary]
```

Arborym commented on auto_comps.py:
```diff
@@ -0,0 +1,224 @@
+        deck_list.append((nms, afs))
+    for nms, afs in deck_list:
+        deck_keys = parse.parse_qs(parse.urlparse(afs).query)['deck'][0]
+        deck_response = requests.get("https://lolchess.gg/builder/set8?hl=en&deck=" + deck_keys)
```
f-strings! 😃 
```suggestion
        deck_response = requests.get(
            f"https://lolchess.gg/builder/set8?hl=en&deck={deck_keys}"
        )
```

Arborym commented on auto_comps.py:
```diff
@@ -0,0 +1,224 @@
+        deck_keys = parse.parse_qs(parse.urlparse(afs).query)['deck'][0]
+        deck_response = requests.get("https://lolchess.gg/builder/set8?hl=en&deck=" + deck_keys)
+        pattern = r'<script id="__NEXT_DATA__" type="application/json">\s*({[\s\S]*?})\s*</script>'
+        json_in_text = re.search(pattern, deck_response.text).group(1)
```
Grouping unnecessary here
```suggestion
        json_in_text = re.search(pattern, deck_response.text)[1]
```

Arborym commented on auto_comps.py:
```diff
@@ -0,0 +1,224 @@
+                    champion_name = \
+                    list(filter(lambda e: e['key'] == each_slot.get("champion"), query_data.get("champions")))[
+                        0]['name']
+                except:
```
```suggestion
                except Exception:
```

Arborym commented on auto_comps.py:
```diff
@@ -0,0 +1,224 @@
+    json_parsed = json.loads(loaded_string.text)
+    set_data_champions = json_parsed["setData"][2]["champions"]
+    output = {}
+    for each_champion in set_data_champions:
+        board_size = 1
+        traits = each_champion["traits"]
+        for each_trait in traits:
+            if each_trait == 'Dragon':
+                board_size = 2
+                break
+        name_tft = each_champion["apiName"]
+        ingame_name = each_champion["name"]
+        if name_tft not in output:
+            output[name_tft] = (ingame_name, board_size)
+    return output
```
Simplifying and refactoring some of the logic here:
```suggestion
    for each_champion in set_data_champions:
        traits = each_champion["traits"]
        board_size = next((2 for each_trait in traits if each_trait == 'Dragon'), 1)
        name_tft = each_champion["apiName"]
        if name_tft not in output:
            ingame_name = each_champion["name"]
            output[name_tft] = (ingame_name, board_size)
    return output
```

Arborym commented on auto_comps.py:
```diff
@@ -0,0 +1,224 @@
+        with open(cached_file_path + set_current + '.json', 'w') as f:
+            f.write(jsoned_champions + '\n')
+            f.write(jsoned_comps)
+    print('Set: ' + set_current + ', loaded champions: ' + str(len(comp_manager.champions)) + ', comps: ' + str(
+        len(comp_manager.comps_loaded)))
+    for i in range(0, len(comp_manager.comps_loaded)):
+        temp = ','.join(comp_manager.comps_loaded[i][1])
+        print(str(i) + ' - ' + comp_manager.comps_loaded[i][0] + ' [' + temp + ']')
```
More f-strings! 😃 (also removed an unnecessary 0 in range)
```suggestion
    print(
        f'Set: {set_current}, loaded champions: {len(comp_manager.champions)}, comps: {len(comp_manager.comps_loaded)}'
    )
    for i in range(len(comp_manager.comps_loaded)):
        temp = ','.join(comp_manager.comps_loaded[i][1])
        print(f'{str(i)} - {comp_manager.comps_loaded[i][0]} [{temp}]')
```

Arborym commented on auto_comps.py:
```diff
@@ -0,0 +1,224 @@
+            temp_inputed = f.read()
+        if temp_inputed != '':
+            print(
+                'Your last selection was: "' + temp_inputed + '", press Enter to use last selection, or type "n" and press Enter to make new selection')
+            if input().lower() == 'n':
+                inputed = ''
+            else:
+                inputed = temp_inputed
+
```
f-string + if expression instead of statement
```suggestion
                f'Your last selection was: "{temp_inputed}", press Enter to use last selection, or type "n" and press Enter to make new selection'
            )
            inputed = '' if input().lower() == 'n' else temp_inputed
```

Arborym commented on auto_comps.py:
```diff
@@ -0,0 +1,224 @@
+        comp_manager.is_sequence_mode = False
+    elif inputed == 'all':
+        comp_manager.is_sequence_mode = True
+        for i in range(0, len(comp_manager.comps_loaded)):
```
```suggestion
        for i in range(len(comp_manager.comps_loaded)):
```

Arborym commented on auto_comps.py:
```diff
@@ -0,0 +1,224 @@
+    elif inputed.startswith('all_except'):
+        comp_manager.is_sequence_mode = True
+        splitted = inputed.split(' ')
+        temp_except = []
+        for each in splitted:
+            if each.isnumeric():
+                inted = int(each)
+                temp_except.append(inted)
+        for i in range(0, len(comp_manager.comps_loaded)):
```
list comprehension on `temp_except`
```suggestion
        temp_except = [int(each) for each in splitted if each.isnumeric()]
        for i in range(len(comp_manager.comps_loaded)):
```

Arborym commented on comps.py:
```diff
@@ -1,148 +1,58 @@
-]
+    def SetCOMPSLoaded(self, input):
+        self.comps_loaded = input
+    def SelectNextComp(self):
+        sequence_len = len(self.sequence)
+        comps_size = len(self.comps_loaded)
+        if self.is_sequence_mode == False:
+            self.index_current = random.randint(0, comps_size-1)
+        else:
+            self.sequence_index = self.sequence_index + 1
+            if (self.sequence_index >= sequence_len):
+                self.sequence_index = 0
+            self.index_current = self.sequence[self.sequence_index]
+        print(f"[!] {self.CURRENT_COMP()[0]} [{','.join(self.CURRENT_COMP()[1])}] comp is selected")
```
Rearrange logic
```suggestion
    def SelectNextComp(self):
        if self.is_sequence_mode == False:
            comps_size = len(self.comps_loaded)
            self.index_current = random.randint(0, comps_size-1)
        else:
            self.sequence_index = self.sequence_index + 1
            sequence_len = len(self.sequence)
            if (self.sequence_index >= sequence_len):
                self.sequence_index = 0
            self.index_current = self.sequence[self.sequence_index]
        print(f"[!] {self.CURRENT_COMP()[0]} [{','.join(self.CURRENT_COMP()[1])}] comp is selected")
```

Arborym commented on game.py:
```diff
@@ -137,8 +142,34 @@ def pvp_round(self) -> None:
+            """Level to 5 at 2-5"""
+            while arena_functions.get_level() < 5:
+                self.arena.buy_xp_round()
+            else:
+                print(f"\n[LEVEL UP] Lvl. {arena_functions.get_level()}")
```
Don't need any else's here
```suggestion
            print(f"\n[LEVEL UP] Lvl. {arena_functions.get_level()}")
```

Arborym commented on game.py:
```diff
@@ -137,8 +142,34 @@ def pvp_round(self) -> None:
+            """Level to 6 at 3-2"""
+            while arena_functions.get_level() < 6:
+                self.arena.buy_xp_round()
+            else:
+                print(f"\n[LEVEL UP] Lvl. {arena_functions.get_level()}")
```
```suggestion
            print(f"\n[LEVEL UP] Lvl. {arena_functions.get_level()}")
```

Arborym commented on game.py:
```diff
@@ -137,8 +142,34 @@ def pvp_round(self) -> None:
+            """Level to 7 at 4-1"""
+            while arena_functions.get_level() < 7:
+                self.arena.buy_xp_round()
+            else:
+                print(f"\n[LEVEL UP] Lvl. {arena_functions.get_level()}")
```
```suggestion
            print(f"\n[LEVEL UP] Lvl. {arena_functions.get_level()}")
```

Arborym commented on game.py:
```diff
@@ -137,8 +142,34 @@ def pvp_round(self) -> None:
+            """Level to 8 at 5-1"""
+            while arena_functions.get_level() < 8:
+                self.arena.buy_xp_round()
+            else:
+                print(f"\n[LEVEL UP] Lvl. {arena_functions.get_level()}")
```
```suggestion
            print(f"\n[LEVEL UP] Lvl. {arena_functions.get_level()}")
```

danparizher reviewed (changes requested):
Great work implementing this! I think this is done very well. I took the time and added some of my insight for refactoring and cleaning up the code, comments attached. 😄

Sizzzles committed:
refactoring and code clean up

Co-authored-by: Daniel Parizher <105245560+Arborym@users.noreply.github.com>

Sizzzles committed:
refactoring and code clean up

Co-authored-by: Daniel Parizher <105245560+Arborym@users.noreply.github.com>

Sizzzles committed:
refactoring and code clean up

Co-authored-by: Daniel Parizher <105245560+Arborym@users.noreply.github.com>

Sizzzles committed:
refactoring and code clean up

Co-authored-by: Daniel Parizher <105245560+Arborym@users.noreply.github.com>

Sizzzles committed:
refactoring and code clean up

Co-authored-by: Daniel Parizher <105245560+Arborym@users.noreply.github.com>

Sizzzles committed:
refactoring and code clean up

Co-authored-by: Daniel Parizher <105245560+Arborym@users.noreply.github.com>

Sizzzles committed:
refactoring and code clean up

Co-authored-by: Daniel Parizher <105245560+Arborym@users.noreply.github.com>

Sizzzles committed:
refactoring and code clean up

Co-authored-by: Daniel Parizher <105245560+Arborym@users.noreply.github.com>

Sizzzles committed:
refactoring and code clean up

Co-authored-by: Daniel Parizher <105245560+Arborym@users.noreply.github.com>

Sizzzles committed:
refactoring and code clean up

Co-authored-by: Daniel Parizher <105245560+Arborym@users.noreply.github.com>

Sizzzles committed:
refactoring and code clean up

Co-authored-by: Daniel Parizher <105245560+Arborym@users.noreply.github.com>

Sizzzles committed:
refactoring and code clean up

Co-authored-by: Daniel Parizher <105245560+Arborym@users.noreply.github.com>

Sizzzles committed:
Update auto_comps.py

Co-authored-by: Daniel Parizher <105245560+Arborym@users.noreply.github.com>

Sizzzles committed:
Update auto_comps.py

Co-authored-by: Daniel Parizher <105245560+Arborym@users.noreply.github.com>

Sizzzles committed:
Update auto_comps.py

Co-authored-by: Daniel Parizher <105245560+Arborym@users.noreply.github.com>

Sizzzles committed:
Update auto_comps.py

Co-authored-by: Daniel Parizher <105245560+Arborym@users.noreply.github.com>

Sizzzles committed:
Update auto_comps.py

Co-authored-by: Daniel Parizher <105245560+Arborym@users.noreply.github.com>

Sizzzles committed:
Update auto_comps.py

Co-authored-by: Daniel Parizher <105245560+Arborym@users.noreply.github.com>

Sizzzles committed:
Update auto_comps.py

Co-authored-by: Daniel Parizher <105245560+Arborym@users.noreply.github.com>

Sizzzles commented:
> Great work implementing this! I think this is done very well. I took the time and added some of my insight for refactoring and cleaning up the code, comments attached. 😄

Thank you a lot for having a look means a lot! Added all your suggestions. Again thank you and I appreciate you!

Sizzzles committed:
Fixed auto comp augments not being added

Sizzzles committed:
Comp augments cleanup

Sizzzles committed:
Removed Combat Training duplicate

Sizzzles committed:
increased sleep between clicks for move_unknown

Sizzzles committed:
Set 8.5 update

Sizzzles committed:
Fixed early loss.... (Maybe?)

Sizzzles committed:
Merge branch 'main' into main

Sizzzles committed:
updates comps by deleting old versions on launch

Sizzzles committed:
Delayed auto queue after game

Sizzzles committed:
delayed move_unnkown, wasnt placing some units

Sizzzles committed:
Merge branch 'main' of https://github.com/Sizzzles/TFT-OCR-BOT

Sizzzles committed:
Merge branch 'main' into main

Sizzzles committed:
Types /ff since button disabled

Sizzzles committed:
Merge branch 'main' of https://github.com/Sizzzles/TFT-OCR-BOT

Sizzzles committed:
Updated main.py

Sizzzles committed:
10 sec queue delay to fix ghost match

Sizzzles committed:
Bel'Veth fixed

Sizzzles committed:
Stopped level up spam

Sizzzles committed:
move_known and move_unknown have same delay

Sizzzles committed:
Decreased tacticians crown check sleep

Sizzzles committed:
Updated anvil, item placement rounds

Sizzzles committed:
Pickup items timing reverted, removed select_shop

Sizzzles committed:
Update arena.py

Sizzzles committed:
Added second round, moved bench cleanup order

Sizzzles committed:
Removed take all button, adjusted item pick loc

Sizzzles committed:
Merge branch 'main' into main

Sizzzles committed:
Second round fix

Sizzzles committed:
Move champion delay changed

Sizzzles committed:
Item checks after carousels

danparizher committed:
Refactorings and optimizations

Sizzzles committed:
Merge branch 'main' into main

Sizzzles committed:
Merge pull request #14 from Arborym/main

Refactorings and optimizations

Sizzzles committed:
Played with timings to move units

Sizzzles committed:
Comp update confirmation

Sizzzles/TFT-OCR-BOT#10

Sizzzles committed:
Revert "Comp update confirmation"

This reverts commit e04ef26438a94d252d5419400ec17c736377ed7e.

Sizzzles committed:
Comp update confirmation

Sizzzles/TFT-OCR-BOT#11

Sizzzles committed:
Update README.md

Sizzzles committed:
Update README.md

Sizzzles committed:
Moved comp update confirmation to main.py

This should stop the bot asking if you want to update the comps twice

Sizzzles committed:
Fixed accidental deletion of auto comps selector

Sizzzles committed:
Possible fix for level up looping
Sizzzles/TFT-OCR-BOT#16

Sizzzles committed:
Fix for incorrect Nunu name usage in comps

Sizzzles committed:
Update main.py

Sizzzles committed:
Item pickup coords update

Sizzzles committed:
Item order fix, added Tacticians Crown for portals
Sizzzles/TFT-OCR-BOT#10

Sizzzles committed:
Update README.md

Sizzzles committed:
Working Set 9 Version

Sizzzles committed:
Working Set 9 Version

Sizzzles committed:
Working Set 9 Version

Sizzzles committed:
Update issue templates

Sizzzles committed:
Region Augment and Augment Rerolls update

Sizzzles committed:
Added Portal round, delay for region augment

- Changed round 1-1 to portal round (tactician now stands still)
- Added delay when checking for Region Augment

Sizzzles committed:
Second round starting delay

- Added delay at start of second round so unit is moved to board correctly

Sizzzles committed:
Improvements for Set 9

- Corrected Ryze's name for auto_comps
- Added portal voting (Uses same system as augments. Change your desired portals in game_assets)

Sizzzles committed:
Update README.md

Sizzzles committed:
Merge pull request #18 from Sizzzles/Sizzzles-patch-1

Update README.md

Sizzzles committed:
Create Close Stale Issues..yml

Sizzzles committed:
Update Close Stale Issues..yml

Sizzzles committed:
Update Close Stale Issues..yml

Sizzzles committed:
Delete Close Stale Issues..yml

Sizzzles committed:
Changes to unit levels, assets, item pickups

- Region Augment Marus Omegnum pickup location
- Auto comps unit levels matche lolchess instead of forcing 3 stars
    augments from game_assets no longer in use (pulls from lolchess)
- Updated items in game_assets (reformatting)
- Faster item pickup (Thanks @kizxlove)
- Updated dependencies

Sizzzles committed:
Merge branch 'main' of https://github.com/Sizzzles/TFT-OCR-BOT

Sizzzles committed:
update pylint

Sizzzles committed:
.gitignore update

Sizzzles committed:
Refactors and whitespace removal

danparizher committed:
Small Refactor

Sizzzles committed:
Merge pull request #22 from Arborym/main

Small Refactor

Sizzzles committed:
Improvements

- Matching OG bot changed pytesseract to tesserocr, python 3.11.4, requirements updates, readme updates
- Stillwater region augment implementation
- Exit button works again since riot moved it

Sizzzles committed:
9.5 Update

Sizzzles committed:
Update README.md

Sizzzles committed:
comp builder imports set number now

Sizzzles committed:
Merge branch 'main' of https://github.com/Sizzzles/TFT-OCR-BOT

Sizzzles committed:
updated .gitignore

Sizzzles committed:
Fix - Level added to summon units, TF name fix

Sizzzles committed:
Match OG bot

- Checks if unknown champ is in current comp
- Better augment reroll code and coord updates

Sizzzles committed:
Set 10 Update, Code Refactoring
Sizzzles/TFT-OCR-BOT#26

Sizzzles committed:
Use game_assets + lolchess augments
1. using game_assets augments while waiting for lolchess to include augments in comps

Sizzzles committed:
Fix Akali naming, always picking augments (PvE)

Sizzzles committed:
Added headliners, use undefined items and fixes

Sizzzles committed:
Update issue templates

Sizzzles committed:
Update issue templates

Sizzzles committed:
Fixed comp retrieval, improved logic, readability

- Improved item handling for champions (still needs work)
- Updated champion name and price fetching from LOLCHESS (until they break it again...) Sizzzles/TFT-OCR-BOT#28
- Added Fast 8 leveling logic for high value comp
- Streamlined various function arguments and return statements
- Code restructuring for improved readability and formatting

Sizzzles committed:
"removed" debug prints

notsapinho commented:
I guess there is no headliner support for now, any problems if I try to implement it?

notsapinho commented:
Are headliners working?

Sizzzles commented:
@notsapinho They should be. If not make an issue on my page

notsapinho commented:
@Sizzzles changing headliner text positions to the following seemed to fix the issue.

```py
HEADLINER_POS: list[Vec4] = [
    Vec4(GameWindow(1280, 984, 1294, 982)),
    Vec4(GameWindow(1280, 964, 1316, 1004)),
    Vec4(GameWindow(1280, 1014, 1294, 1028)),
    Vec4(GameWindow(1280, 1014, 1316, 1030))
]
```

Sizzzles committed:
Avoid Augments/Clearer Headliner replacement

Sizzzles committed:
Comps load again..... For now
Sizzzles/TFT-OCR-BOT#30

Sizzzles committed:
Improve checking in queue and fix avoid augments

Sizzzles committed:
Added avoid portals and changed some screen_coords

Sizzzles committed:
Fix: Keeping carousel Illaoi if in comp

Changes made:
Fixed Illaoi's name being read as "Ilaoi"

Sizzzles committed:
Update requirements.txt typo

Sizzzles committed:
Tacticians Crown logic improved, added retry to auto_queue

Sizzzles committed:
Merge branch 'main' of https://github.com/Sizzzles/TFT-OCR-BOT

Sizzzles committed:
Removed Tacticians Crown debug logging

Sizzzles committed:
Set 11 update (probably broke something)

Sizzzles committed:
Fix set 11 update crash (rip headliners)

Sizzzles committed:
Removed board debugs

Sizzzles committed:
Set 11 update: Augment coords and item names

Sizzzles committed:
Fix champ name key errors

Sizzzles committed:
Removed all headliner code, fixed unit names
Sizzzles/TFT-OCR-BOT#37

Sizzzles committed:
Encounter rounds and fixed rek'sai name

Sizzzles committed:
Forgot to update CRAFTABLE_ITEMS_DICT

Sizzzles committed:
Forgot space for Lee Sin

Sizzzles committed:
Trying to get dummies to work + fix_bench_state for Xayah Rakan

Sizzzles committed:
Update Encounter Rounds + Pick Coords

Sizzzles committed:
Use of champion duplicators and CUSTOM COMPS

Sizzzles committed:
Fix duplicator crash and Added trait emblem logic

Sizzzles committed:
Trait check for Trait emblem logic

Sizzzles committed:
Item pick coords change

Sizzzles committed:
Fix Kayle crash in duplicators and improve glove placement on wrong champions

Sizzzles committed:
Match og bot

- setting up for possible item placement recode
